[RFC] lantiq: add support for VGV953 (Speedport W 921V)

Wingman2k opened the pull request:
Create VGV953.dts
Update platform.sh, Makefile and 02_network

there is no wifi module for XWAVE300
possible bugs with nand and usb when checking dmesg
USB works and I did not had any issues with the partitions yet
LEDs and Network Ports are all mapped
2nd kernel line present in dts file with sram commented out
Old Install instruction is still valid
https://openwrt.org/toh/t-com/spw921v
known u-boot size limitation for initramfs image
usb power with gpio 32 high so assume dummy voltage regulator

Signed-off-by: Stefan Grau <stefan_grau@t-online.de>

hailfinger commented:
(minor nitpick) Your commit refers to the old wiki. Could you please link to the new wiki instead? Just remove the "wiki." part of the URL. Thanks.

hailfinger commented:
Merging this patch would also resolve the ancient (from 2015) pull request for this device:
https://github.com/openwrt-mirror/openwrt/pull/15

rmilecki commented on target/linux/lantiq/files-4.14/arch/mips/boot/dts/VGV953.dts:
```diff
@@ -0,0 +1,337 @@
+/dts-v1/;
```
Please include licensing info for your DTS file, see
https://openwrt.org/submitting-patches#dts_checklist
e.g. a line like:
`// SPDX-License-Identifier: GPL-2.0-or-later OR MIT`
above the `/dts-v1/;`

hailfinger commented:
Does this device use the default NAND badblocks marker, or does it use the modified badblocks marker like the Easybox 904 xDSL?

adrianschmutzler commented:
This has formal issues, needs rebase and has never been updated by the original author.
Please address the remark, otherwise we will close it.

Wingman2k commented:
The requested changes are possible, when flashed on the device and configured it revealed heavy performance issues which i could not resolve, it should be more responsive than the TD-W8970 but device rebooted and never worked stable on the DSL Line, nobody will be happy with such a device.
@hailfinger sorry seems i missed your comment, is there a simple way to figure this out?

adrianschmutzler commented:
@Wingman2k Thanks for your response and for being honest about issues connected to your support patches.

So, I'd propose the following:
If you feel that you will solve the problems in the foreseeable future, keep this PR open and update the PR with fixed content and address the formal issues.

If you come to the decision that more work is required, maybe try to start a discussion in the forum and close this PR. Feel free to reopen if your patches are "ready".
It's also okay to submit an almost-ready device if you comment about the shortcomings and apply the appropriate indicator in your PR title (e.g. RFC).

I'm sorry if this sounds a little harsh, but I haven't found a way to convey this information in written form without that.

Plonkbong commented:
Please let it open, it will need as discussion based.
This devise is in addition to the Easybox 904xDSL the only second, very cheap (secondhand) VDSL-telefon-Router with enough recources.
This device was inside openwrt in 15.05
But now the wifi WAVE300 chipset get working now.

Unfortunately the uboot does not work with the modern openwrt. Did somewhere know where the old uboot sourcecode for this device are is ?

adrianschmutzler commented:
Since there has not been any progress here both concerning updating the PR as well as providing useful help or review, I will close this now (it was opened more than a year ago.)

Feel invited to reopen it (ask for reopening) when there is new development in this context.

Wingman2k commented:
@Plonkbong Updated the wiki with a new u-boot.asc and bin with 16M size limitation.
and original git commit for u-boot and also precompiled files.
@adrianschmutzler image compiles for 19.07.0 and master, but still massive performance issues.
this is on ext-root via USB and initramfs and nand (loading software list never finish)
file is almost rebased, try to push something later, device is busy now.
@hailfinger with "nand-on-flash-bbt" the system crashes on boot

I would like to reopen, as RFC with the new u-boot files testing is easier now.
the device also self reboots sometimes and drops into u-boot after that.

adrianschmutzler commented:
@Wingman2k I have reopened it.

Wingman2k commented:
Thanks, meanwhile i tried the master build, one error message i need to resolve first:
```
pinctrl-xway 1e100b10.pinmux: pin io42 already requested by 1e100b10.pinmux; cannot claim for 1e108000.eth
pinctrl-xway 1e100b10.pinmux: pin-42 (1e108000.eth) status -22
pinctrl-xway 1e100b10.pinmux: could not request pin 42 (io42) from group mdio  on device 1e100b10.pinmux
lantiq,xrx200-net 1e108000.eth: Error applying setting, reverse things back
lantiq,xrx200-net: probe of 1e108000.eth failed with error -22

```
Performance might be CPU, dropbear keygen with 4096 bits takes 1 minute on BT-HH5A
over 7 on this device

Wingman2k commented:
Full Board Name is VGV953AKW22-B-23 is it OK to use VGV953
USB Node 0 is only used to get io32 to high, is there a better way?

adrianschmutzler commented:
> Full Board Name is VGV953AKW22-B-23 is it OK to use VGV953

I'm not a fan of abbreviated device names, at some point they always cause confusion because they because ambiguous.

Wingman2k committed:
[RFC] lantiq: add support for VGV953 (Speedport W 921V) #1477

Rebase and Update files, new device name VGV953AKW22
the B is Annex Type, like ARV8539PW22, we dont care
performance seems OK removed phy-leds and exim
Updated u-boot version and some Files at
https://openwrt.org/toh/t-com/spw921v
Device Features not (currently) working:
WIFI, Phone (Analog, ISDN, Dect)
Cleaned up some empty lines and squashed

Signed-off-by: Stefan Grau stefan_grau@t-online.de

Wingman2k commented:
will add exim3 and stp to nodes again, LED are not working without it.
change of partition kernel to 4M, currently 2M kernel is close to limit.

Wingman2k committed:
4M kernel size and stp + exim3 added
will be squashed later i guess
missed on old device name in vr9.mk

Wingman2k committed:
led device names changed from vgv953 to vgv953akw22

Wingman2k commented on target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_arcadyan_vgv953akw22.dts:
```diff
@@ -0,0 +1,312 @@
+	lantiq,groups = <0x7>;
+	lantiq,dsl = <0x0>;
+	lantiq,phy1 = <0x0>;
+	lantiq,phy2 = <0x0>;
```
this could be removed?

Wingman2k commented on target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_arcadyan_vgv953akw22.dts:
```diff
@@ -0,0 +1,312 @@
+&usb_phy0 {
+	status = "okay";
+};
+
```
Not used for USB but to get pin io32 on high, better way?

Wingman2k commented on target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_arcadyan_vgv953akw22.dts:
```diff
@@ -0,0 +1,312 @@
+&usb1 {
+	status = "okay";
+	vbus-supply = <&usb1_vbus>;
+};
```
Not used for USB but to get pin io32 on high, better way?

Wingman2k commented on target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_arcadyan_vgv953akw22.dts:
```diff
@@ -0,0 +1,312 @@
+
+		gpio = <&gpio 32 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
```
See USB comment further down only needed for "gpio = <&gpio 32 GPIO_ACTIVE_HIGH>;"

andyboeh replied:
Why do you need the GPIO to be high?
I'm just commenting here, because I had a similar problem on the AVM FRITZ!3390. I added a `gpio-hog` to force the GPIO to a predefined state, but on the Lantiq you need one more modification to `vr9.dtsi` which is also included in #2662, but not yet merged.

Wingman2k replied:
GPIO 32 and 33 are necessary to power the USB Ports, only usb_phy1 with 2 Ports.
If one of the GPIO's is high the port is not supplied with 5V.
it looks nicer with the gpio-hog, found your commit earlier but overlooked it. Thanks for mentioning it, this requires either your commit to be merged or added to this dts file.
ifxhcd-rst-dev or ifxhcd-pwr would be an appropriate name?

Wingman2k committed:
gpio hog

Wingman2k commented:
flashing a new uboot is currently required it was mentioned somewhere this is not preferred
there might be a possibility to not need this, but then some flash is not usable
it has a dual image design with 2x ~4-5M for kernel and 16M for the Voice recorder which could be used for overlay

Plonkbong commented:
Hi there,
it is possible to make the FXS ports usable ?
I try to add the nosmp bootargs and add &vmmc
see: https://github.com/Plonkbong/openwrt/blob/lantiq-xrx200-next1/target/linux/lantiq/files-4.19/arch/mips/boot/dts/lantiq/vr9_arcadyan_vgv953akw22-b-23-vpe.dts
but without success
It is the code from Easybox 904xDSL. I have no idea what are the right gpio number is for the slic.
Any ideas ?

adrianschmutzler commented:
Hi, since adding device support is essentially a feature, not a fix, and none of the committer seems to be interested in reviewing it, I will close it now (i.e. "Won't implement").

I'm sorry, but better crush your remaining hopes now than waiting another year and wasted further development time of yours where supposedly nothing will be different.
